query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/NewQueryResource.scala

```diff
@@ -180,14 +182,20 @@ class NewQueryResource(
         }))
 
         var allSecondaries: Stream[Option[String]] =
-          analysis.statement.allTables.toStream.scanLeft((Option.empty[String], Set.empty[String])) { (acc, dss) =>
-            val (_, alreadySeen) = acc
-            val DatabaseTableName((DatasetInternalName(n), Stage(s))) = dss
-            secondary.chosenSecondaryName(None, n, Some(s), alreadySeen) match {
-              case Some(secondary) => (Some(secondary), alreadySeen + secondary)
-              case None => (None, alreadySeen)
-            }
-          }.map(_._1).tail
+          store match {
+            case None =>
+              analysis.statement.allTables.toStream.scanLeft((Option.empty[String], Set.empty[String])) { (acc, dss) =>
+                val (_, alreadySeen) = acc
+                val DatabaseTableName((DatasetInternalName(n), Stage(s))) = dss
+                secondary.chosenSecondaryName(None, n, Some(s), alreadySeen) match {
+                  case Some(secondary) => (Some(secondary), alreadySeen + secondary)
+                  case None => (None, alreadySeen)
+                }
+              }.map(_._1).tail
+            case Some(store) =>
+              log.info("Forcing secondary to {}", JString(store))
```

Comment:
Is there a reason for wrapping it in a JString here? Quotes?

Reply:
Yep.  It's a user-provided string, and I prefer to not just dump stuff like that into the log without any kind of escaping.